sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -993,6 +994,14 @@ class CodegenContext {
     }
   }
 
+  /**
+   * Returns the code for subexpression elimination after splitting it if necessary.
+   */
+  def subexprFunctionsCode: String = {
+    // Whole-stage codegen's subexpression elimination is handled in another code path
+    splitExpressions(subexprFunctions, "subexprFunc_split", Seq("InternalRow" -> INPUT_ROW))
```

Comment:
We don't need to check if its empty?
```
if (subexprFunctions.nonEmpty) {
  splitExpressions(...
} else {
 ""
}
```

Reply:
It is not necessary: if it is empty `splitExpressions` would return an empty string, anyway I can add it if you think it is more clear